omod/src/main/java/org/openmrs/module/appui/fragment/controller/SessionFragmentController.java

```diff
@@ -47,7 +47,11 @@ public List<SimpleObject> getLoginLocations(UiUtils ui, @SpringBean AppFramework
         List<Location> loginLocations = appFrameworkService.getLoginLocations();
         List<SimpleObject> ret = new ArrayList<SimpleObject>();
         for (Location location : loginLocations) {
-            ret.add(SimpleObject.fromObject(location, ui, "id", "uuid", "name"));
+            SimpleObject obj = new SimpleObject();
+            obj.put("id", location.getId());
+            obj.put("uuid", location.getUuid());
+            obj.put("name", ui.format(location));
+            ret.add(obj);
```

Comment:
This makes sense to me.  If it is important that `name` remain the actual name on the Location for some use cases, an alternative would be to first construct the SimpleObject as before, and then add in an additional `display` property that is the value of `ui.format(location)`.  This would also be consistence (I think) with how the REST module operates.  Obviously this is more involved, as you'd then need to change any pages/fragments to use the `display` instead of `name` when rendering, but it's something to consider.